src/tsickle.ts

```diff
@@ -776,11 +778,18 @@ class Annotator extends ClosureRewriter {
         // If it has a symbol, it's actually a regular declared property.
         if (!quotedPropSym) return false;
         const propName = (eae.argumentExpression as ts.StringLiteral).text;
-        // Properties containing non-JS identifier names must not be unquoted.
+        // Properties containing non-JS identifier names can only be accessed with quotes.
         if (!isValidClosurePropertyName(propName)) return false;
-        this.writeNode(eae.expression);
-        this.emit(`.${propName}`);
-        return true;
+        const symName = this.typeChecker.symbolToString(quotedPropSym);
+        this.debugWarn(
```

Comment:
I think we have to move it to a linter if we actually want developers to be aware of the issue.
I can get the type-aware linting back up and running today.